pkg/internal/codegen/parse/crd.go

```diff
@@ -402,6 +404,26 @@ func (b *APIs) parseArrayValidation(t *types.Type, found sets.String, comments [
 	return props, buff.String()
 }
 
+// parseInterfaceValidation returns a JSONSchemaProps object and its serialization in
+// Go that describe the validations for the given Interface.
+func (b *APIs) parseInterfaceValidation(t *types.Type, found sets.String, comments []string) (v1beta1.JSONSchemaProps, string) {
+	props := v1beta1.JSONSchemaProps{
+		OneOf: []v1beta1.JSONSchemaProps{
+			{
+				Type: "string",
+			},
+			{
+				Type: "integer",
+			},
+			{
+				Type: "object",
```

Comment:
why should an interface be no slice? no number? Feels wrong. We shouldn't invent semantics which is not generic.

Reply:
@sttts yeah, my bad. I will rectify this error.